libs/bulldozer/client/applications/components/application-explorer/src/lib/application-explorer.component.ts

```diff
@@ -1,91 +1,107 @@
+
+            <bd-collection-explorer
+              [connected]="connected"
+              [applicationId]="application.id"
+              [workspaceId]="workspaceId"
             >
-              <mat-icon>more_horiz</mat-icon>
-            </button>
-          </div>
-        </mat-expansion-panel-header>
+            </bd-collection-explorer>
 
-        <bd-collection-explorer
-          [connected]="connected"
-          [applicationId]="application.id"
-          [workspaceId]="workspaceId"
-        >
-        </bd-collection-explorer>
+            <bd-instruction-explorer
+              [connected]="connected"
+              [applicationId]="application.id"
+              [workspaceId]="workspaceId"
+            >
+            </bd-instruction-explorer>
 
-        <bd-instruction-explorer
-          [connected]="connected"
-          [applicationId]="application.id"
-          [workspaceId]="workspaceId"
-        >
-        </bd-instruction-explorer>
+            <mat-menu #applicationOptionsMenu="matMenu">
+              <a
+                mat-menu-item
+                [routerLink]="[
+                  '/workspaces',
+                  application.data.workspace,
+                  'applications',
+                  application.id
+                ]"
+              >
+                <mat-icon>launch</mat-icon>
+                <span>View application</span>
+              </a>
+              <button
+                mat-menu-item
+                bdEditApplicationTrigger
+                [application]="application"
+                (editApplication)="onUpdateApplication(application.id, $event)"
+                [disabled]="!connected"
+              >
+                <mat-icon>edit</mat-icon>
+                <span>Edit application</span>
+              </button>
+              <button
+                mat-menu-item
+                (click)="onDeleteApplication(application.id)"
+                [disabled]="!connected"
+              >
+                <mat-icon>delete</mat-icon>
+                <span>Delete application</span>
+              </button>
+            </mat-menu>
+          </mat-expansion-panel>
+        </div>
 
-        <mat-menu #applicationOptionsMenu="matMenu">
-          <a
-            mat-menu-item
-            [routerLink]="[
-              '/workspaces',
-              application.data.workspace,
-              'applications',
-              application.id
-            ]"
-          >
-            <mat-icon>launch</mat-icon>
-            <span>View application</span>
-          </a>
-          <button
-            mat-menu-item
-            bdEditApplicationTrigger
-            [application]="application"
-            (editApplication)="onUpdateApplication(application.id, $event)"
-            [disabled]="!connected"
-          >
-            <mat-icon>edit</mat-icon>
-            <span>Edit application</span>
-          </button>
+        <div
+          id="bd-application-buttons"
```

Comment:
is there a reason to use an id selector for this? I think this just sets this div a height of 100px, it might be easier to maintain to just use a tailwind class